transports/tcp/src/lib.rs

```diff
@@ -368,7 +371,25 @@ where
         socket.bind(&socket_addr.into())?;
         socket.listen(self.config.backlog as _)?;
         socket.set_nonblocking(true)?;
-        TcpListenStream::<T>::new(id, socket.into(), self.port_reuse.clone())
+        let listener: TcpListener = socket.into();
+        let local_addr = listener.local_addr()?;
+
+        if local_addr.ip().is_unspecified() {
```

Comment:
shouldn't we check if port is unspecified too?

Reply:
That was my first thought as well. But while an unspecified IP results in one IP per (potentially future) interface, an unspecified port results in a **single** randomly chosen port.

@elenaf9 @melekes please correct me in case I am missing something here.

Reply:
> an unspecified port results in a single randomly chosen port.

Well, I am not sure what the OS does when it can not find a unique port number that is free on all interfaces. I am guessing that it will choose a different port number per interface. Though still **only one** per interface.

Reply:
> That was my first thought as well. But while an unspecified IP results in one IP per (potentially future) interface, an unspecified port results in a single randomly chosen port.

Yes exactly.